### DISCUSSION TITLE
Form conditions based on BelongTO

### DISCUSSION THREAD

**Discussion by mrahmadt** [filamentphp/filament#8385] [Help] [answered]
Hello

If I have **ModelA** and in the Create/Edit Form, I will have a select input that will be the name from **ModelB**

Now I want the other inputs in this form to have some conditions to be available in the form or not 

**Example:**
ModelA = Book
ModelB = Author

**Fields/Inputs in Create/Edit Form:**
field_select[ModelA.author_id]  -> select/show ModelB.name
ModeA.acceptBids ---> Only if ModelB.allowResell is true

Any idea how to get ModelB data when selecting it from the select field? and how to use it with other fields in ModelA?

Thanks

**Comment by leandrocfe** [comment]
[dependant-fields](https://filamentphp.com/docs/3.x/forms/getting-started#dependant-fields) ?

**Reply by mrahmadt** [reply]
Thanks @leandrocfe  but this will work for the fields in the current Model, it will not allow me to get field from other Model

**Reply by leandrocfe** [reply]
You can use relationships

**Comment by mrahmadt** [comment, marked as answer]
This is what I managed to get so far; not sure if this is the best solution


```
                Select::make('modelb_id')
                ->relationship(name: 'modelb', titleAttribute: 'name')
                ->loadingMessage('Loading...')->live()
                ->afterStateUpdated(function (Set $set, Get $get) {
                    $ModelB = ModelB::where(['id'=>$get('modelb_id'), 'isActive'=>true ])->first();
                    $set('note', $ModelB->note);
                }),
                TextInput::make('note')->columnSpan('full')->required(),
```
